parquet_collator.py

```diff
@@ -0,0 +1,103 @@
+import asyncio
+import logging
+import urllib.parse
+
+import boto3
+import pandas as pd
+import yaml
+
+from measuring_ci.utils import find_staged_data_files
+
+LOG_LEVEL = logging.INFO
+
+# AWS artisinal log handling, they've already set up a handler by the time we get here
+log = logging.getLogger()
+log.setLevel(LOG_LEVEL)
+# some modules are very chatty
+logging.getLogger("aiohttp").setLevel(logging.INFO)
+
+
+def load_parquet(filename, columns):
+    """Load existing parquet file or an empty one."""
+    try:
+        df = pd.read_parquet(filename)
+        log.info("Loaded %s", filename)
+    except Exception as exc:
+        log.info("Couldn't load %s, using empty data set (%s)", filename, exc)
+        df = pd.DataFrame(columns=columns)
+    return df
+
+
+def delete_parquet_files(url_list):
+    """Remove a series of s3 objects, which must be in the same bucket."""
+    s3_client = boto3.client('s3')
+    bucket = urllib.parse.urlparse(url_list[0]).netloc
+    delete_data = {
+        'Objects': [{'Key': urllib.parse.urlparse(p).path.lstrip('/')} for p in url_list],
+    }
+
+    s3_client.delete_objects(Bucket=bucket, Delete=delete_data)
+
+
+async def collate_parquet_files(args, config):
+    """Collect parquet data files into one."""
+    # Don't store the full path to integration/releases/...
+    if 'project' in args:
+        short_project = args['project'].split('/')[-1]
+        config['total_cost_output'] = config['total_cost_output'].format(project=short_project)
+        config['staging_output'] = config['staging_output'].format(project=args['project'])
+    staged_files = await find_staged_data_files(config['staging_output'])
+
+    # Without this, urllib doesn't know about s3 and removes s3://netloc/
+    urllib.parse.uses_netloc.append('s3')
+    urllib.parse.uses_relative.append('s3')
```

Comment:
it looks like these affect the global urllib.parse state? maybe they should be moved up into the top-level module?